docs/_docs/reference/other-new-features/generalized-method-syntax.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: doc-page
+title: "Generalized Method Syntax"
+movedTo: https://docs.scala-lang.org/scala3/reference/other-new-features/generalized-method-syntax.html
+---
+
+The inclusion of using clauses is not the only way in which methods have been updated, type parameter clauses are now allowed in any number and at any position.
+
+## Syntax Changes
+
+### In Scala 2
+
+The old syntax only allowed zero or one type parameter clause, followed by any number of term clauses, optionnally followed by an implicit clause:
+
+```scala
+def foo[T, U](x: T)(y: U)(z: Int, s: String)(using Ord[Int])(a: Array[T])(implicit l: List[U])
+```
+
+### In Scala 3
+
+The new syntax allows any number of type clauses, as long as they are not adjacent:
+(do note however that [implicit clause are discouraged, in favor of using clauses](https://docs.scala-lang.org/scala3/reference/contextual/relationship-implicits.html))
+
+```scala
+def foo[T, U](x: T)(y: U)[V](z: V, s: String)(using Ord[Int])[A](a: Array[A])(implicit List[U])
+```
+
+### Unchanged
+
+Class definitions and type declarations are unaffected, there can only be up to one type clause, in leading posion.
+
+## Motivation
+
+The new syntax is a powerful but natural extension of the old one, it allows new design patterns while staying intuitive and legible.
+
+### Dependent Type Clauses
+
+As type clauses can come after term clauses, it is now possible to have type parameters depend on term parameters:
+
+```scala
+trait Key { type Value }
+trait DB {
+  def get(k: Key): Option[k.Value] // dependent return type
+  def getOrElse(k: Key)[V >: k.Value](default: V): V // dependent type parameter
+}
+```
+
+Note that simply replacing `V` by `k.Value` would not be equivalent. For example, if `k.Value` is `Int`, the above allows: 
+`getOrElse(k)[Number](0.1)`, which returns a `Number`.
```

Comment:
`Number` is not a supertype of `Int`. It's only a supertype of `Integer`. You need a different example here.

Reply:
I think I'll use `Option` then, thank you for reminding me